modular_skyrat/master_files/code/modules/antagonists/traitor/objectives/kill_pet.dm

```diff
@@ -0,0 +1,39 @@
+/datum/traitor_objective/kill_pet
+	possible_heads = list(
+		JOB_HEAD_OF_PERSONNEL = list(
+			/mob/living/simple_animal/pet/dog/corgi/ian,
+			/mob/living/simple_animal/pet/dog/corgi/puppy/ian
+		),
+		JOB_CAPTAIN = /mob/living/simple_animal/pet/fox/renault,
+		JOB_CHIEF_MEDICAL_OFFICER = /mob/living/simple_animal/pet/cat/runtime,
+		JOB_CHIEF_ENGINEER = /mob/living/simple_animal/parrot/poly,
+		// Non-heads like the warden, these are automatically medium-risk at minimum
+		JOB_ROBOTICIST = /mob/living/simple_animal/pet/dog/corgi/borgi,
+		JOB_STATION_ENGINEER = /mob/living/simple_animal/pet/poppy,
+	)
+
+/datum/traitor_objective/kill_pet/generate_objective(datum/mind/generating_for, list/possible_duplicates)
+	. = ..()
+	// Dust Poppy, the safety inspector
+	if(istype(target_pet, /mob/living/simple_animal/pet/poppy))
+		name = "Dust the engineering department's esteemed safety inspector and beloved pet, Poppy"
+		description = "A couple of troublemakers in the engineering department have spilled the milk, make them and their colleagues pay for the consequences by throwing Poppy, the safty inspector into the Super Matter engine. "
+
+		for(var/datum/component/traitor_objective_register/objective as anything in GetComponents(/datum/component/traitor_objective_register))
+			if(objective.target == target_pet)
+				qdel(objective)
+
+		AddComponent(/datum/component/traitor_objective_register, target_pet, \
+			succeed_signals = list(COMSIG_PARENT_QDELETING)) // Until dusting gets its own component, this has to make do
+
+	// Emag E-N so it overloads
```

Comment:
As much as I like the objective in theory, is there a precedent for forcing someone to buy a specific item for an objective?

Reply:
Thank you for the review 😊 There isn't as far as I know, but since this is so new I figured it would be fun to try.
Emags are locked behind a rep gate and you can only get one by the time you would also get this objective, with the large amount of starting TC the chance is high that you get an emag anyway.

Reply:
Why not just provide a single-use emag for this

Reply:
Actually that's exactly what you should do. You can use the battlecruiser emag as a reference for how to make this.

Reply:
Yeah I was thinking something like that. See if you can make a single-use emag for the purpose.